kairon/shared/chat/agent/agent_flow.py

```
@@ -77,12 +77,14 @@ def create_fake_tracker(self, slot_vals: dict[str,any] = None, sender_id: str =
         if self.should_use_global_slots:
             global_slots = self.load_global_slots()
             for slot in slots:
-                if slot.name in global_slots:
+                if (not slot_vals or slot.name not in slot_vals.keys()) and (slot.name in global_slots):
```

Ruff (0.8.2)

Use `key not in dict` instead of `key not in dict.keys()`

Remove `.keys()`

(SIM118)